lib/routes/shared/initial_walkthrough.dart

```diff
@@ -63,16 +70,23 @@ class InitialWalkthroughPageState extends State<InitialWalkthroughPage>
                 new RestoreDialog(context, widget._backupBloc, options));
       }
 
-      String restorePIN;
+      String restoreKey;
       if (toRestore != null) {
         if (toRestore.encrypted) {
-          restorePIN = await getRestorePIN();
-          if (restorePIN == null) {
+          if (toRestore.encryptionType == "Mnemonics") {
+            restoreKey = await getBackupPhrase();
+            if (restoreKey != null) _updateSecurityModelAction = UpdateSecurityModel(widget._user.securityModel.copyWith(backupKeyType: BackupKeyType.PHRASE));
+          } else if (toRestore.encryptionType == "Pin") {
+            restoreKey = await getRestorePIN();
+            if (restoreKey != null) _updateSecurityModelAction = UpdateSecurityModel(widget._user.securityModel.copyWith(backupKeyType: BackupKeyType.PIN));
```

Comment:
At both cases where the restoreKey is given the `backupKeyType` is changes correctly but the key is not saved in the secure storage.

Reply:
[Fixed](https://github.com/breez/breezmobile/pull/173/commits/17d400a70e8decc8aada4810b6d092c33b2f8054)